Testing/SystemTests/lib/systemtests/systemtesting.py

```diff
@@ -706,8 +708,11 @@ def start_in_current_process(self, script):
             detail = ex.args[0]
             cl, exc, tb = sys.exc_info()
             line_number = traceback.extract_tb(tb)[-1][1]
-        print(f"{error_class} at line {line_number} of SystemTest for {script._modname}.{script._test_cls_name}: "
-              f"{detail}")
+        except SystemExit as ex: # catch sys.exit
+            exitcode=ex.args[0]
+        if line_number:
+            print(f"{error_class} at line {line_number} of SystemTest for {script._modname}.{script._test_cls_name}: "
+                  f"{detail}")
```

Comment:
The line number I get output on a single thread test failure is incorrect (refers to a file in unittest, rather than in ISIS_PowderPolarisTest.py). On multithreaded, the failure prints a full stacktrace, which is clearer! I think this gets printed in [testThreadsLoop L1390 in systemtesting.py](https://github.com/mantidproject/mantid/blob/main/Testing/SystemTests/lib/systemtests/systemtesting.py#L1388-L1391)